apis/Google.Cloud.Logging.Log4Net/Google.Cloud.Logging.Log4Net/GoogleStackdriverAppender_Configuration.cs

```diff
@@ -177,62 +201,96 @@ public void AddResourceLabel(Label label)
 
         /// <summary>
         /// Specify additional metadata to include in all log entries.
         /// </summary>
         /// <param name="enable">The additional metadata to enable.</param>
-        public void AddWithMetaData(MetaDataType enable)
-        {
-            _withMetaData.Add(enable);
-        }
+        public void AddWithMetaData(MetaDataType enable) => _withMetaData.Add(ThrowIfActivated(enable, nameof(AddWithMetaData)));
 
+        private int _serverErrorBackoffDelaySeconds = 1;
         /// <summary>
         /// On receiving a server error during log upload, the initial delay in seconds before retry.
         /// Defaults value is 1 second.
         /// </summary>
-        public int ServerErrorBackoffDelaySeconds { get; set; } = 1;
+        public int ServerErrorBackoffDelaySeconds
+        {
+            get => _serverErrorBackoffDelaySeconds;
+            set => _serverErrorBackoffDelaySeconds = ThrowIfActivated(value, nameof(ServerErrorBackoffDelaySeconds));
+        }
 
+        private double _serverErrorBackoffMultiplier = 1.5;
         /// <summary>
         /// The multiplier applied to the retry delay when receiving multiple consecutive server errors during log upload.
         /// Default value is 1.5
         /// </summary>
-        public double ServerErrorBackoffMultiplier { get; set; } = 1.5;
+        public double ServerErrorBackoffMultiplier
+        {
+            get => _serverErrorBackoffMultiplier;
+            set => _serverErrorBackoffMultiplier = ThrowIfActivated(value, nameof(ServerErrorBackoffMultiplier));
+        }
 
+        private int _serverErrorBackoffMaxDelaySeconds = 60;
         /// <summary>
         /// The maxmimum retry delay when receiving multiple consecutive server errors during log upload.
         /// Default value is 60 seconds.
         /// </summary>
-        public int ServerErrorBackoffMaxDelaySeconds { get; set; } = 60;
+        public int ServerErrorBackoffMaxDelaySeconds
+        {
+            get => _serverErrorBackoffMaxDelaySeconds;
+            set => _serverErrorBackoffMaxDelaySeconds = ThrowIfActivated(value, nameof(ServerErrorBackoffMaxDelaySeconds));
+        }
 
+        private int _disposeTimeoutSeconds = 30;
         /// <summary>
         /// The maximum time the <c>Dispose()</c> call of <see cref="GoogleStackdriverAppender"/> may take. 
         /// Default value is 30 seconds.
         /// </summary>
-        public int DisposeTimeoutSeconds { get; set; } = 30;
+        public int DisposeTimeoutSeconds
+        {
+            get => _disposeTimeoutSeconds;
+            set => _disposeTimeoutSeconds = ThrowIfActivated(value, nameof(DisposeTimeoutSeconds));
+        }
+
+        private T ThrowIfActivated<T>(T value, string name)
+        {
+            GaxPreconditions.CheckState(!_isActivated, $"Appender already activated; cannot modify '{name}'");
```

Comment:
Tiny nit:

```csharp
GaxPreconditions.CheckState(!_isActivated, "Appender already activated; cannot modify '{0}'", name);
```

... that avoids the string formatting for non-errors. Admittedly this is unlikely to be a performance bottleneck, but...